consensus/src/common.rs

```diff
@@ -30,6 +30,47 @@ where
     fn state_hash(&self) -> Option<StateHash>;
 }
 
+impl Chain<ProtocolState> for ProtocolState {
```

Comment:
I don't see why we need to implement Chain here on ProtocolState, as it's already implemented on `ProtocolStateChain`. Also these API's only make sense on a blockchain. ProtocolState is a just a single block header, so by definition it should not be implementing Chain trait.

Reply:
Ok, I will remove `ProtocolState` from  `Chain `trait.